### PR TITLE
Create `dex.pools` Macros & Move to Sector - Ready for review

### PR DESCRIPTION
**Pull request by henrystats**
This pr adds dex.pools to the _sector/dex category.

I've set up macros that can be reusable by uniswap forks so it's easier to add new projects to the dex.pools model.

The code used in the macro was written by @hildobby [here](https://github.com/duneanalytics/spellbook/pull/5007#issuecomment-1971295693) and I've asked for his permission to reuse it in this pr for the macros. 

The code also allows for addition of pools with more than 2 tokens to dex.pools

### CODE REVIEW

**Comment by henrystats** [comment]
Hola  ser @Hosuke, can you please help here :(, I've been trying to fix this for the past couple of hours and I keep running into the same "Null entry" issue, I've re-written the macros multiple times to try and find what is causing the issue but it's just not working. 

dbt compile is running fine and the compiled sql is also running fine on dune.com

**Comment by Hosuke** [comment]
Not sure the exact cause yet, still debugging.

**Comment by henrystats** [comment]
@Hosuke I found the issue in the original PR that Hildobby set up, so everything works correctly, thank you very much ser!

@jeff-dude this is now ready for review and I wanted to also ask what approach to take for adding the existing dexes to this sector, since a lot of them are forks of uniswap, it'd be pretty straightforward, there's a lot!

I can submit them In batches right? (maybe by chain, so uniswap dexes on optimism, uniswap dexes on ethereum etc etc)

After I add the existing fantom pools, I can remove the previous dex_pools and change the current one from dex_pools_beta to dex_pools.

File: macros/models/_sector/dex/uniswap_compatible_pools.sql [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    {% if hardcoded_fee %} -- use hardcoded fee if it's exists
+    , {{ hardcoded_fee }} as fee
+    {% endif %}
+    {% if fee_column_name %}
+    , f.{{fee_column_name}} as fee -- use fee column if hardcoded fee doesn't exists
+    {% endif %}
+    {% if not (fee_column_name or hardcoded_fee) %}
+    , CAST(NULL as BIGINT) as fee 
+    {% endif %}
+    , array_agg(
```
Comment (Hosuke) [inline macros/models/_sector/dex/uniswap_compatible_pools.sql:28]: Is it possible to extend this `tokens` column into `token_symbols`?

**Review by Hosuke — APPROVED** [review]
LGTM.✅
Thank you @henrystats 

[Sample uniswap_v2_ethereum_pools]()
[Sample uniswap_v3_ethereum_pools]()

File: models/_sector/dex/pools/dex_pools_beta.sql
Comment (Hosuke) [inline models/_sector/dex/pools/dex_pools_beta.sql:1]: We may create an issue for the `dex.pools_beta` to track all related PRs.

File: macros/models/_sector/dex/uniswap_compatible_pools.sql (hunk 1 above)
Comment (henrystats) [reply macros/models/_sector/dex/uniswap_compatible_pools.sql:28]: hmm sure @Hosuke, I will add it in

Comment (henrystats) [reply macros/models/_sector/dex/uniswap_compatible_pools.sql:28]: token_symbols added @Hosuke

**Comment by henrystats** [comment]
bump @jeff-dude 👀

**Pull request merged**